include/seqan3/core/simd/simd_algorithm.hpp

```diff
@@ -77,6 +293,289 @@ constexpr simd_t iota(typename simd_traits<simd_t>::scalar_type const offset)
     return detail::iota_impl<simd_t>(offset, std::make_integer_sequence<scalar_type, length>{});
 }
 
+/*!\brief Load simd_t size bits of integral data from memory.
+ * \ingroup simd
+ * \tparam    simd_t   The simd type; must model seqan3::simd::Simd.
+ * \param[in] mem_addr The memory address to load from. Does not need to be aligned on any particular boundary.
+ *
+ * \details
+ *
+ * \include test/snippet/core/simd/simd_load.cpp
+ */
+template <Simd simd_t>
+//!\cond
+    requires detail::is_builtin_simd_v<simd_t> && (detail::is_native_builtin_simd_v<simd_t>)
+//!\endcond
+constexpr simd_t load(void const * mem_addr)
+{
+    assert(mem_addr != nullptr);
+
+    if constexpr (simd_traits<simd_t>::max_length == 16)
+        return reinterpret_cast<simd_t>(_mm_loadu_si128(reinterpret_cast<__m128i const *>(mem_addr)));
+    else if constexpr (simd_traits<simd_t>::max_length == 32)
+        return reinterpret_cast<simd_t>(_mm256_loadu_si256(reinterpret_cast<__m256i const *>(mem_addr)));
+    else if constexpr (simd_traits<simd_t>::max_length == 64)
+        return reinterpret_cast<simd_t>(_mm512_loadu_si512(mem_addr));
+    else
+        static_assert(simd_traits<simd_t>::max_length >= 16 && simd_traits<simd_t>::max_length <= 64,
+                      "Unsupported simd type.");
+}
+
+//!\overload
+template <Simd simd_t>
+//!\cond
+    requires detail::is_builtin_simd_v<simd_t> && (!detail::is_native_builtin_simd_v<simd_t>)
+//!\endcond
+constexpr simd_t load(void const * mem_addr)
```

Comment:
I only do it because it seems that evaluating a constexpr if seems to be faster than resolving overloads by evaluating concepts. Otherwise, I find it more readable.

Reply:
I think this tag dispatching style is not better readable (even if you use requires to do it). this is so seqan2 style. :chart_with_downwards_trend: